awx/ui_next/src/components/LaunchPrompt/useLaunchSteps.js

```diff
@@ -8,51 +9,59 @@ import usePreviewStep from './steps/usePreviewStep';
 export default function useSteps(config, resource, i18n) {
   const [visited, setVisited] = useState({});
   const steps = [
-    useInventoryStep(config, resource, visited, i18n),
-    useCredentialsStep(config, resource, visited, i18n),
-    useOtherPromptsStep(config, resource, visited, i18n),
-    useSurveyStep(config, resource, visited, i18n),
+    useInventoryStep(config, visited, i18n, resource),
+    useCredentialsStep(config, i18n),
+    useOtherPromptsStep(config, i18n),
+    useSurveyStep(config, visited, i18n),
   ];
+  const { values: formikValues, errors } = useFormikContext();
 
-  const formErrorsContent = steps
-    .filter(s => s?.formError && Object.keys(s.formError).length > 0)
-    .map(({ formError }) => formError);
+  const formErrorsContent = [];
 
+  if (config.ask_inventory_on_launch && !formikValues.inventory) {
+    formErrorsContent.push({
+      inventory: true,
+    });
+  }
+  const hasSurveyError = Object.keys(errors).find(e => e.includes('survey'));
+
+  if (
+    config.survey_enabled &&
+    (config.variables_needed_to_start ||
+      config.variables_needed_to_start.length === 0) &&
+    hasSurveyError
+  ) {
+    formErrorsContent.push({
+      survey: true,
+    });
+  }
```

Comment:
Is there a reason this validation got moved out of each individual step? Does it differ between different prompt wizards?